course-mw/sunbird-util/sunbird-cassandra-utils/src/main/java/org/sunbird/cassandraimpl/CassandraOperationImpl.java

```diff
@@ -736,4 +740,50 @@ public Response searchValueInList(
     return response;
   }
 
+  @Override
+  public Response batchInsertLogged(
+          RequestContext requestContext, String keyspaceName, String tableName, List<Map<String, Object>> records) {
+
+    long startTime = System.currentTimeMillis();
+    logger.info(requestContext,
+            "Cassandra Service batchInsertLogged method started at ==" + startTime);
+
+    Session session = connectionManager.getSession(keyspaceName);
+    Response response = new Response();
+    BatchStatement batchStatement = new BatchStatement(BatchStatement.Type.LOGGED);
+    batchStatement.setConsistencyLevel(ConsistencyLevel.QUORUM);
+    ResultSet resultSet = null;
+    List<String> writeType = new ArrayList<String>(){{add(WriteType.BATCH.name());add(WriteType.SIMPLE.name());}};
+
+    try {
+      for (Map<String, Object> map : records) {
+        Insert insert = QueryBuilder.insertInto(keyspaceName, tableName);
+        map.entrySet()
+                .stream()
+                .forEach(
+                        x -> {
+                          insert.value(x.getKey(), x.getValue());
+                        });
+        batchStatement.add(insert);
+      }
+      resultSet = session.execute(batchStatement);
+      response.put(Constants.RESPONSE, Constants.SUCCESS);
+    } catch (QueryExecutionException
+            | QueryValidationException
+            | NoHostAvailableException
+            | IllegalStateException e) {
+      logger.info(requestContext, "Cassandra Batch Insert Failed." + e.getMessage(), e);
```

Comment:
Only QueryExecutionException should be caught, and WriteTimeoutException should be handled properly